network/vpack/parse.go

```diff
@@ -0,0 +1,253 @@
+// Copyright (C) 2019-2025 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package vpack
+
+import "fmt"
+
+type voteValueType uint8
+
+const (
+	credPfVoteValue voteValueType = iota
+	rPerVoteValue
+	rPropDigVoteValue
+	rPropEncdigVoteValue
+	rPropOperVoteValue
+	rPropOpropVoteValue
+	rRndVoteValue
+	rSndVoteValue
+	rStepVoteValue
+	sigP1sVoteValue
+	sigP2VoteValue
+	sigP2sVoteValue
+	sigPVoteValue
+	sigPsVoteValue
+	sigSVoteValue
+)
+
+func parseVote(data []byte, c *StatelessEncoder) error {
```

Comment:
need a comment stating `parseVote` parses msgp and produces encoding vote by using `StatelessEncoder`

Reply:
Maybe rename `data` to `msgpacked`?